tests/store/tracking/test_file_store_logged_model.py

```diff
@@ -0,0 +1,619 @@
+    store.set_logged_model_tags(logged_model.model_id, [LoggedModelTag("tag_key", "new_tag_value")])
+    logged_model = store.get_logged_model(logged_model.model_id)
+    assert logged_model.tags == {"tag_key": "new_tag_value"}
+    store.set_logged_model_tags(
+        logged_model.model_id, [LoggedModelTag("a", None), LoggedModelTag("b", 123)]
+    )
+    logged_model = store.get_logged_model(logged_model.model_id)
+    assert logged_model.tags == {"tag_key": "new_tag_value", "a": "", "b": "123"}
+
+
+def test_set_logged_model_tags_errors(store):
+    logged_model = store.create_logged_model()
+    with pytest.raises(MlflowException, match=r"Missing value for required parameter"):
+        store.set_logged_model_tags(logged_model.model_id, [LoggedModelTag(None, None)])
+    with pytest.raises(MlflowException, match=r"Names may only contain alphanumerics"):
+        store.set_logged_model_tags(logged_model.model_id, [LoggedModelTag("a!b", "c")])
+
+
+def test_get_logged_model(store):
+    experiment_id = store.create_experiment("test")
+    run_id = store.create_run(
+        experiment_id,
+        user_id="user",
+        start_time=0,
+        tags=[],
+        run_name="test_run",
+    ).info.run_id
+    logged_model = store.create_logged_model(
+        experiment_id=experiment_id,
+        name="test_model",
+        source_run_id=run_id,
+        tags=[LoggedModelTag("tag_key", "tag_value")],
+        params=[LoggedModelParameter("param_key", "param_value")],
+        model_type="dev",
+    )
+    fetched_model = store.get_logged_model(logged_model.model_id)
+    assert logged_model.model_uri == fetched_model.model_uri
+    assert logged_model.to_dictionary() == fetched_model.to_dictionary()
+
+
+def test_get_logged_model_errors(store):
+    with pytest.raises(MlflowException, match=r"Model '1234' not found"):
+        store.get_logged_model("1234")
+
+    with (
+        mock.patch(
+            "mlflow.store.tracking.file_store.FileStore._find_model_root",
+            return_value=("0", "abc"),
+        ),
+        mock.patch(
+            "mlflow.store.tracking.file_store.FileStore._get_model_info_from_dir",
+            return_value={"experiment_id": "1"},
+        ),
+    ):
+        with pytest.raises(MlflowException, match=r"Model '1234' metadata is in invalid state"):
+            store.get_logged_model("1234")
+
+
+def test_finalize_logged_model(store):
+    logged_model = store.create_logged_model()
+    assert logged_model.status == str(LoggedModelStatus.PENDING)
+    logged_model_dict = logged_model.to_dictionary()
+    updated_model = store.finalize_logged_model(logged_model.model_id, LoggedModelStatus.READY)
+    assert updated_model.status == str(LoggedModelStatus.READY)
+    updated_model_dict = updated_model.to_dictionary()
+    for k in logged_model_dict:
+        if k not in ["status", "last_updated_timestamp"]:
+            assert logged_model_dict[k] == updated_model_dict[k]
```

Comment:
```suggestion
    assert logged_model.status == updated_model.status
    assert logged_model.last_updated_timestamp == updated_model.last_updated_timestamp
```


Do we need to use `to_dictionary`?

Reply:
I see, can we remove the loop?